pyronear/datasets/wildfire/wildfire.py

```diff
@@ -158,10 +158,10 @@ def fit(self, wildfire):
         """
         self.wildfire = wildfire
         # Some checks first
-        if wildfire.metadata['fire_id'].nunique() != wildfire.metadata['fire_id'].max() + 1:
-            warnings.warn(f"Inconsistent Fire Labeling. Maybe try to label the fire again\n"
-                          f"Distinct values of ids({wildfire.metadata['fire_id'].nunique()}"
-                          f" ≠ {wildfire.metadata['fire_id'].max() + 1})", Warning)
+        # if wildfire.metadata['fire_id'].nunique() != wildfire.metadata['fire_id'].max() + 1:
+        #     warnings.warn(f"Inconsistent Fire Labeling. Maybe try to label the fire again\n"
+        #                   f"Distinct values of ids({wildfire.metadata['fire_id'].nunique()}"
+        #                   f" ≠ {wildfire.metadata['fire_id'].max() + 1})", Warning)
```

Comment:
If commented, do we need to keep this section at all?

Reply:
This warning is here for some reason. Splitting this dataset suppose firelabeling was correctly made.
If it wasn't, a warning is raised to warn the user. It could be difficult to see it from outside the function.

Besides, If you want to silence warnings when using this code in your script/notebook in case too much information displayed bother you, you can do it this way: (no need to silence warnings for everyone)
```python
import warnings
warnings.filterwarnings("ignore")
```

Reply:
For me a warning that is raised under any circumstances doesn't really make sense. We can eventually put it back on if we annotate the entire dataset. We can also modify it by considering what was actually annotated

Reply:
Indeed, It does raise a warning under any circumstances corresponding to the same misuse of metadata.

If you instantiate a `WildFireDataset` object with a metadata file, it should cover exactly the values of this dataset, not more not less. Here a warning is raised because such inconsistency was detected.It is important, because this a a responsibility for the splitter not to mess the fire ids. 

Also, the count of frames in the directory used to init WildFireDataset does not prevail on the metadata to give information on the dataset (actually, this is the role of _metadata_ to do so).
This is even more dangerous because using a metadata not reflecting the data, give _wrong lenght of WildFireDataset !_ so we can expect that some splitting strategies may process the dataset in unexpected ways !
Following this conversation, I think precising this in the documentation and testing metadata consistency over the data it covered in WildFireDataset may help us.

So, If you work on a subset of Wildfire today, it may be better working with a subset of the metadata.csv, and it should work well. At some point you should have made this distinction.

Reply:
Okay, I'm sorry, I think I don't understand something. I'm gonna explain my thinking to you to see if I'm missing something. I added the FireLabels to my dataset, the fires are labeled from the names of the 1020 videos, which gives 420 different fires, which is what we find in the warning:

`wildfire.metadata['fire_id'].max()`

But since we didn't annotate all the dataset the number of fireid actually present in the annotations is 340.  `wildfire.metadata['fire_id'].nunique()`
What raised the warning:

`Distinct values of ids(340 ≠ 420)`

Am I using the fireLabeler in the wrong way? Should I have deleted the unlabeled videos from my video list?

Reply:
Thank you for detailing your workflow, I understand now more clearly you're trying to achieve.

I recommend to only work with the data available. Checks should pass and it would be more convenient for you to work. Here is a suggestion.

Let's see the flow [from the videos to the WildFireDataset](https://user-images.githubusercontent.com/6331909/77332236-182adf80-6d22-11ea-897d-6449a02a77fa.png)

At beginning, we can have:
- all videos in a directory
- checkedstates **covering only annotated data**
- metadata _covering all videos_ <-- this needs to be changed to **cover only annotated data**

So if you, create a new metadata.csv from the original(exhaustive) one, you tell the other objects (WildFireDataset, FireLabeler, WildFiresplitter...) the videos you are going to use.

The role of metadata is to _provide information of the data (videos/frames) you need to process_. The other objects will process data _base on these information_, so that's important metadata only contain what's needed/available.
Concerning the videos/frames, you shouldn't need to subselect what's needed/delete unannotated videos because the objects(splitter...) will be able to filter based on metadata.csv.

Of course, we can improve this process. It is very valuable to have your feedback, tell me if this helps.
Thanks,

Reply:
Okay, I'll work on my workflow. I'll put the warning back on. Thanks